src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -245,16 +260,30 @@ CreateReferenceTable(Oid relationId)
 						errdetail("There are no active worker nodes.")));
 	}
 
+	/* we only support data loading for regular (non-foreign) relations */
+	relationKind = get_rel_relkind(relationId);
+	if (relationKind == RELKIND_RELATION)
+	{
+		canLoadData = true;
+	}
+
 	colocationId = CreateReferenceTableColocationId();
 
 	/* first, convert the relation into distributed relation */
 	ConvertToDistributedTable(relationId, distributionColumnName,
-							  DISTRIBUTE_BY_NONE, REPLICATION_MODEL_2PC, colocationId);
+							  DISTRIBUTE_BY_NONE, REPLICATION_MODEL_2PC, colocationId,
+							  canLoadData);
```

Comment:
Left in the logic to copy only on regular relations, but got rid of this variable in favor of a single `requireEmpty` variable for this argument.

The call to load data just directly checks the relation kind.